redwood-protocol-guest/src/jsMain/kotlin/app/cash/redwood/protocol/guest/PlatformList.kt

```diff
@@ -20,6 +20,12 @@ internal external class JsArray<E> {
   @JsName("length")
   val size: Int
 
+  @JsName("at")
+  operator fun get(index: Int): E
+
+  @JsName("set")
```

Comment:
I aught to validate that this works before I merge this PR.

Reply:
For `set` I used `splice`: https://github.com/JakeWharton/platform-collections/blob/trunk/src/jsMain/kotlin/com/jakewharton/platformcollections/PlatformList.kt#L56-L58

No clue why I did this. Does Kotlin even let you produce `subject[index] = value` in the JS? There's no normal function to use other than bracket notation, right? I didn't see a `set`.

For `get` I used `asDynamic()` plus regular indexing: https://github.com/JakeWharton/platform-collections/blob/trunk/src/jsMain/kotlin/com/jakewharton/platformcollections/PlatformList.kt#L28-L34

This saves a few bytes and looks more idiomatic in the JS. `at` is relatively new, but should be fine for browsers and QuickJS though.

Not sure if you want index checking. Brackets and `at` behave differently between -1 and -length. Both return undefined outside that and normal range. An explicit index check means you avoid the negative index behavior of `at`.

Probably want extensions for both operators funs and then at or brackets + splice on the actual type.

Reply:
Extensions worked.